scripts/ffcv/create_ffcv_datasets.py

```diff
@@ -100,12 +111,33 @@ def parse_args():
     return args
 
 
+def cacheStreaming(remote, local, num_workers=64):
+    if "LOCAL_WORLD_SIZE" not in os.environ:
+        os.environ["LOCAL_WORLD_SIZE"] = 1
+
+    ds = StreamingImageClassDataset(remote=remote,
+                                    local=local,
+                                    shuffle=True,
+                                    transform=transforms.CenterCrop(224),
+                                    batch_size=512)
+
+    dl = torch.utils.data.DataLoader(ds,
+                                     batch_size=512,
+                                     num_workers=num_workers,
+                                     pin_memory=False,
+                                     drop_last=False,
+                                     collate_fn=pil_image_collate)
+
+    # empty loop to download and cache the dataset
+    for _ in dl:
```

Comment:
I think this is begging for a blocking StreamingDataset.download() method (future PR)?

Reply:
Yes. This helper function would be useful and it can take arguments like how many workers to use for download.